src/features/create-quiz/create-quiz-first-card/create-quiz-first-card.tsx

```diff
@@ -1,43 +1,217 @@
+  const navigate = useNavigate();
+
+  const toast = useRef<Toast>(null);
+
+  const [quizName, setQuizName] = useState("");
+
+  const handleCancel = () => {
+    QuizStore.deselectQuiz();
+    navigate("/homepage");
+  };
+
+  const showConfirm = () => {
+    toast.current?.show({
+      severity: "warn",
+      sticky: true,
+      content: (
+        <div className="flex flex-column" style={{ flex: "1" }}>
+          <div className="text-center">
+            <i
+              className="pi pi-exclamation-triangle"
+              style={{ fontSize: "3rem" }}
+            ></i>
+            <h4>Are you sure you want to cancel?</h4>
+            <p>Confirm to proceed</p>
+          </div>
+          <div className="grid p-fluid">
+            <div className="col-6">
+              <Button
+                type="button"
+                label="Yes"
+                className="p-button-success"
+                onClick={handleCancel}
+              />
+            </div>
+            <div className="col-6">
+              <Button
+                type="button"
+                label="No"
+                className="p-button-secondary"
+                onClick={() => {
+                  toast.current?.clear();
+                }}
+              />
+            </div>
+          </div>
+        </div>
+      ),
+    });
+  };
+
+  const handleFirebaseCrudMethod = async (methodType: string) => {
+    const quizzesCollectionRef = collection(db, "quizzes");
+
+    try {
+      if (QuizStore.questions.length > 1) {
+        methodType === "add"
+          ? await addDoc(quizzesCollectionRef, {
+              quiz: {
+                quizName,
+                quizQuestions: toJS(QuizStore.questions.slice(0, -1)),
+              },
+              timeStamp: serverTimestamp(),
+            })
+          : await updateDoc(doc(db, "quizzes", QuizStore.selectedQuiz?.id), {
+              quiz: {
+                quizName,
+                quizQuestions: toJS(QuizStore.questions.slice(0, -1)),
+              },
+              timeStamp: serverTimestamp(),
+            });
+        showSuccess(
+          `Quiz has been ${
+            methodType === "add" ? "created" : "updated"
+          } successfuly.`,
+          toast
+        );
+        navigate("/homepage");
+      } else {
+        console.log("No questions");
+        showError("No questions", toast);
+      }
+    } catch (err) {
+      console.log(err);
+      showError("Other error", toast);
+    }
+  };
+
+  const initialValues = {
+    quizName: "",
+  };
+
+  const formik: any = useFormik({
+    initialValues,
+
+    validate: (data) => {
+      let errors: IInputErrors = {} as IInputErrors;
+
+      if (!data.quizName) {
+        errors.quizName = "Quiz name is required.";
+      }
+
+      return errors;
+    },
+    onSubmit: (data) => {
+      if (Object.keys(formik.errors).length === 0) {
+        if (QuizStore.selectedQuizID) {
+          handleFirebaseCrudMethod("update");
```

Comment:
You could extract the method types into a constant and use that instead of plain strings.